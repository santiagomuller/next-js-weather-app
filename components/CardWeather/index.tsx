import { FC, ReactElement } from 'react'
import Router from 'next/router'

interface LocationDataProps{
  city: string,
  regionName: string,
  country: string,
  lon: string,
  lat: string,
  class?: string,
}

interface CardWeatherProps {
  locationData: LocationDataProps,
  temp: number,
}


const CardWeather :FC<CardWeatherProps> = ({locationData, temp}) :ReactElement => {
  const bgImage = locationData.class ? `bg-${locationData.class}` : 'bg-default'
  const styles = `card cursor-pointer text-shadow p-4 mb-4 flex h-32 text-white md:h-64 justify-between items-end ${bgImage}`
  
  const handleClick = () => {
    Router.push(
      `/forecast?lat=${locationData.lat}&lon=${locationData.lon}&region=${locationData.regionName}, ${locationData.country}`,
      `/clima/?lat=${locationData.lat}&lon=${locationData.lon}&region=${locationData.regionName}, ${locationData.country}`,
    )
  }

  return(
    <div className={styles} onClick={handleClick} data-testid={bgImage}>
      <div>
        <h2 className="text-xl family-bold">{locationData?.city}</h2>
        <p>{`${locationData?.regionName}, ${locationData?.country}`}</p>
      </div>
      <div className="text-3xl xl:text-4xl">{temp}°</div>
    </div>
  )
}

export default CardWeather