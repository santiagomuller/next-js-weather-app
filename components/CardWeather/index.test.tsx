import React from 'react';
import renderer from 'react-test-renderer';
import CardWeather from './index';

const mockLocationData = {
  city: 'Caseros',
  regionName: 'Buenos Aires',
  country: 'Argentina',
  lon: '11.20',
  lat: '34.00',
}

it('CardWeather renders without changes', () => {
  const tree = renderer.create(
    <CardWeather locationData={mockLocationData} temp={20} />
  ).toJSON();
  
  expect(tree).toMatchSnapshot();
});