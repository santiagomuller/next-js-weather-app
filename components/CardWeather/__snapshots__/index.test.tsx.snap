// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardWeather renders without changes 1`] = `
<div
  className="card cursor-pointer text-shadow p-4 mb-4 flex h-32 text-white md:h-64 justify-between items-end bg-default"
  data-testid="bg-default"
  onClick={[Function]}
>
  <div>
    <h2
      className="text-xl family-bold"
    >
      Caseros
    </h2>
    <p>
      Buenos Aires, Argentina
    </p>
  </div>
  <div
    className="text-3xl xl:text-4xl"
  >
    20
    Â°
  </div>
</div>
`;
